### ISSUE TITLE
how to generate .npy files?

### ISSUE THREAD

**Issue by daixiangzi** [hx173149/C3D-tensorflow#83]

**Comment by cyh-dev** [comment]
+1 how do you generate it?

**Comment by Totoro-wen** [comment]
hello,Someone knows how to generate an npy file.
thank for your help.

**Comment by rocksyne** [comment]
@Totoro-wen  What really do you want to do? I will give you an example.

```
import numpy as np
x=some_array_I_have_created

np.save(path/to/where/file/hould/be/saved/flename, x)
```

There, you have a numpy file created

**Comment by Totoro-wen** [comment]
thanks.@rocksyne
Firstly,I want to convert this project (or this idea of 3D convolution) into pytorch, so that I can better use it in the aspect of CV. The first time I used tensorlow.
Secondly,Is the npy file generated from training images?This file is useful in the following code, which I cannot fully understand.
`img = img[crop_x:crop_x+crop_size, crop_y:crop_y+crop_size,:] - np_mean[j]`

**Comment by rocksyne** [comment]
@Totoro-wen The purpose of the numpy file is for data [normalization](https://medium.com/@urvashilluniya/why-data-normalization-is-necessary-for-machine-learning-models-681b65a05029). Therefore, the mean calculated over all the training images (frames) which is now stacked as 112x112x3 is saved as a numpy file. 

So in training, each for each extracted frame, which is of shape 112x112x3, the np_mean is extracted to try to bring the matrix values to a somewhat central (0) point so that the values will not be too far apart. [Kindly check this link to if there is any confusion.](https://medium.com/@urvashilluniya/why-data-normalization-is-necessary-for-machine-learning-models-681b65a05029). Cheers!

**Comment by Totoro-wen** [comment]
@rocksyne I see. Thank you very much.

**Comment by hdf1112** [comment]
@rocksyne I don't know how to generate the crop_mean.npy file. Do you know how to generate it?

**Comment by rocksyne** [comment]
@hdf1112 The mean file for this program is already provided. It is called crop_mean.npy

If you are looking at creating a mean file from your own dataset for use, here are the steps; theoretically.

Pseudocode (python style)

```
import numpy as np

frame_stack = list()

for all training video only:
        selected_frame_pile = extract_x_number_of_frames_as_list_from( video )
        frame_stack.append(selected_frame_pile)

mean_val = np.mean(np.array(frame_stack))

# do this if only you need your mean file to be of a certain dimension
mean_val = crop_mean_val(mean_val)

np.save(path/to/where/file/hould/be/saved/filename.npy, mean_val)
```

**Comment by hdf1112** [comment]
@rocksyne Thank you very much.
